core/chain-configs/src/genesis_config.rs

```diff
@@ -337,6 +457,33 @@ impl Genesis {
         hasher.process_genesis(self);
         hasher.finalize()
     }
+
+    pub fn stream_records_with_callback(&self, callback: impl FnMut(StateRecord)) {
+        let reader = BufReader::new(
+            File::open(&self.records_file.path).expect("could not open genesis records file"),
+        );
+        match &self.records_file.file_type {
+            GenesisRecordsFileType::FullGenesis => {
+                stream_records_from_genesis_file(reader, callback);
+            }
+            GenesisRecordsFileType::RecordsArray => {
+                stream_records_from_records_file(reader, callback);
+            }
+        }
+    }
+
+    pub fn process_records(&self, mut callback: impl FnMut(&StateRecord)) {
```

Comment:
```suggestion
    pub fn for_each_record(&self, mut callback: impl FnMut(&StateRecord)) {
```

sounds like a bit more rustic name.